thamos/discover.py

```diff
@@ -71,6 +109,83 @@ def discover_cuda_version(interactive: bool = False) -> typing.Optional[str]:
     return cuda_version
 
 
+def discover_cudnn_version() -> Optional[str]:
+    """Discover CuDNN version and return None if not found."""
+    result = run_command("nvcc --version", raise_on_error=False)
```

Comment:
Why can't the version gathered here be used as the return value?

Reply:
The output of the `nvcc --version` looks something like this:

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Thu_Jun_11_22:26:38_PDT_2020
Cuda compilation tools, release 11.0, V11.0.194
Build cuda_11.0_bu.TC445_37.28540450_0
```

(See https://stackoverflow.com/a/68499241/12700010)

We could try to parse the output string directly but macros for each version number should normally be set in `/usr/local/cuda/include/cudnn.h` or `/usr/include/cudnn.h` if CUDA is installed.

Reply:
Got it. Thanks 👍🏻